backend/organization/models/project.py

```diff
@@ -0,0 +1,150 @@
+from django.db import models
+
+from organization.models import (Organization,)
+
+
+def project_image_path(instance, filename):
+    return "projects/{}/{}".format(instance.id, filename)
+
+
+class Project(models.Model):
+    name = models.CharField(
+        help_text="Points to project name",
+        verbose_name="Name",
+        max_length=1024
+    )
+
+    slug = models.CharField(
+        help_text="URL slug for project",
+        verbose_name="Slug",
+        unique=True,
+        null=True,
+        blank=True,
+        max_length=1024
+    )
+
+    image = models.ImageField(
+        help_text="Project image",
+        verbose_name="Image",
+        null=True,
+        blank=True,
+        upload_to=project_image_path
+    )
+
+    PROJECT_IDEA = "idea"
+    PROJECT_IN_PROGRESS = "in_progress"
+    PROJECT_FINISHED = "finished"
+    PROJECT_CANCELLED = "cancelled"
+    PROJECT_RECURRING = "recurring"
+
+    PROJECT_STATUSES = (
+        (PROJECT_IDEA, "Idea"),
+        (PROJECT_IN_PROGRESS, "In Progress"),
+        (PROJECT_FINISHED, "Finished"),
+        (PROJECT_CANCELLED, "Cancelled"),
+        (PROJECT_RECURRING, "Recurring")
+    )
+
+    status = models.CharField(
+        help_text="Points to status of the project",
+        verbose_name="Status",
+        max_length=64,
+        choices=PROJECT_STATUSES,
+        default=PROJECT_IDEA
+    )
+
+    start_date = models.DateTimeField(
+        help_text="Points to start date of the project",
+        verbose_name="Start Date",
+        null=True,
+        blank=True
+    )
+
+    end_date = models.DateTimeField(
+        help_text="Points to end date of the project",
+        verbose_name="End Date",
+        null=True,
+        blank=True
+    )
+
+    # created_at value automatically added when new project is created
+
+    created_at = models.DateTimeField(
+        help_text="Points to creation date of the project",
+        verbose_name="Created At",
+        auto_now_add=True
+    )
+
+    # updated_at value automatically changes every time there is change in project object.
+    updated_at = models.DateTimeField(
+        help_text="Points to time when project was updated",
+        verbose_name="Updated At",
+        auto_now=True
+    )
+
+    short_description = models.TextField(
+        help_text="Points to short description about the project",
+        verbose_name="Short Description",
+        null=True,
+        blank=True
+    )
+
```

Comment:
Additional simple fields needed would be `country`(TextField), `city`(TextField), `description`(TextField) and `collaborators_welcome`(boolean)

An additional ManyToManyField needed would be `skills` where projects can choose the skills which they could particularly use to accelerate the project. There should be a limit of 5 skills a project can request

An additional array field of freetext TextFields would be helpful_connections. I'm not sure how this can best be implemented. There should be a limit of 5 helpful_connections a project can request.

Reply:
I think limiting skills up to 5 can be done at the API level. Adding that constraint to database level is trickier.

Reply:
Other than limiting skills to 5, all other fields are added.